cmd/clusterctl/client/cluster/mover.go

```diff
@@ -647,18 +654,16 @@ func waitReadyForMove(ctx context.Context, proxy Proxy, nodes []*node, dryRun bo
 		}
 		key := client.ObjectKeyFromObject(obj)
 
-		blockLogged := false
-		if err := retryWithExponentialBackoff(ctx, backoff, func(ctx context.Context) error {
-			if err := c.Get(ctx, key, obj); err != nil {
+		log.Info(fmt.Sprintf("Move blocked by %s annotation, waiting for it to be removed", clusterctlv1.BlockMoveAnnotation))
```

Comment:
I'm not super familiar clusterctl code base, how do you know this annotation is blocking here without checking for their existence?

Reply:
`n.blockingMove` which is checked above is populated earlier in the flow by checking the annotation. This was added by #9246:
https://github.com/kubernetes-sigs/cluster-api/blob/8f82f8850ae689aeacb6398e4e85d4b30595a4e8/cmd/clusterctl/client/cluster/objectgraph.go#L328